nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,53 @@
+const express = require("express");
+const router = express.Router();
+
+const meals = require("./../data/meals.json");
+// const reservations = require("./../data/reservations.json");
+// const reviews = require("./../data/reviews.json");
+
+router.get("/:id", async (request, response) => {
+  const mealId = Number(request.params.id);
+  let mealWithid = meals;
+  try {
+  
+    if(isNaN(mealId)){
+      mealWithid = "You have entered a string. enter a number instead"
+    }
+    if(mealId<=4){
+      mealWithid = meals.find((meal)=> meal.id === Number(request.params.id))}
+    if(mealId>4){
+    mealWithid = "Cannot find any meal for this id"
+    }
```

Comment:
I like that you thought of letting the user know that there are no meals with the id, if it's more than 4(I guess you don't have more than 4 meals) but I don't think you should hardcode the number like that, since you're looking for it in an array I think it'll be best to tell the user, you didn't find it by looking at the array returned which will always be equals 0 right? because with find you'll always get 1 thing returned. So in this case if ``mealWithId`` has a length of 0. you know there was no match.

Reply:
hi thnx Stephen for your feedback really appriciate that. please find below the suggested changes in the new commit. I resolved the hardcoding issue as per your suggestion.